app/fissile.go

```diff
@@ -468,6 +526,27 @@ func (f *Fissile) GenerateRoleImages(targetPath, repository, metricsPath string,
 		return fmt.Errorf("Error loading roles manifest: %s", err.Error())
 	}
 
+	if outputDirectory != "" {
+		err = os.MkdirAll(outputDirectory, 0755)
+		if err != nil {
+			if !os.IsExist(err) {
```

Comment:
```go
if os.IsExist(err) {
   return fmt.Errorf("Output directory exists and is not a directory")
} else if err != nil { 
   return fmt.Errorf("Failed to create output directory")
}
```

I think this is sufficient:
https://golang.org/pkg/os/#MkdirAll
The documentation says that it will not error if the path exists and is a directory. So your only two error conditions is that it exists and it's not a directory, or your hard disk has started a chemical fire.